test_rmw_implementation/test/test_unique_identifiers.cpp

```diff
@@ -113,6 +119,41 @@ TEST_F(CLASSNAME(TestUniqueIdentifierAPI, RMW_IMPLEMENTATION), get_gid_with_bad_
   EXPECT_TRUE(gids_are_equal);
 }
 
+
+TEST_F(CLASSNAME(TestUniqueIdentifierAPI, RMW_IMPLEMENTATION), get_cli_gid_with_bad_args) {
+  rmw_gid_t gid{};
+  rmw_ret_t ret = rmw_get_gid_for_client(cli, &gid);
+  ASSERT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  rmw_gid_t expected_gid = gid;
+
+  ret = rmw_get_gid_for_client(nullptr, &gid);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret);
+  rmw_reset_error();
+  bool gids_are_equal = false;
+  ret = rmw_compare_gids_equal(&expected_gid, &gid, &gids_are_equal);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_TRUE(gids_are_equal);
+
+  ret = rmw_get_gid_for_client(cli, nullptr);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret);
+  rmw_reset_error();
+  gids_are_equal = false;
+  ret = rmw_compare_gids_equal(&expected_gid, &gid, &gids_are_equal);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_TRUE(gids_are_equal);
+
+  const char * implementation_identifier = cli->implementation_identifier;
+  cli->implementation_identifier = "not-an-rmw-implementation-identifier";
+  ret = rmw_get_gid_for_client(cli, &gid);
+  cli->implementation_identifier = implementation_identifier;
+  EXPECT_EQ(RMW_RET_INCORRECT_RMW_IMPLEMENTATION, ret);
+  rmw_reset_error();
+  gids_are_equal = false;
+  ret = rmw_compare_gids_equal(&expected_gid, &gid, &gids_are_equal);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_TRUE(gids_are_equal);
```

Comment:
why?
`gid` should never have changed if the function called change, so it should still be equal to the original `expected_gid`

Reply:
That's correct. It is still equal to the original, hence we expect a true value. This test is checking that `gid` is not modified whenever we make a bad call to the getter. I think this logic is basically copied over from the tests for the publisher getter.

Reply:
ah sorry, I misread the code